includes/class-indieauth-authenticate.php

```diff
@@ -10,6 +10,22 @@ public function __construct() {
 		add_filter( 'determine_current_user', array( $this, 'determine_current_user' ), 11 );
 		add_filter( 'rest_authentication_errors', array( $this, 'rest_authentication_errors' ) );
 		add_action( 'authenticate', array( $this, 'authenticate' ) );
+		// Remove Micropub adding same headers
+		if ( ! defined( 'MICROPUB_LOCAL_AUTH' ) ) {
```

Comment:
IndieAuth is the main authority for authorization and authentication and MicroPub only implements a fallback, so it should be the other way around. IndieAuth should set a const and MicroPub should check it.

Reply:
for whether micropub uses the indieauth plugin or not? sure, micropub can look for a constant that indieauth defines.

we still want micropub to be usable without the indieauth plugin, thought, and ideally still let users decide whether to use indieauth.com or wordpress login, so we'd probably keep this constant for when the indieauth plugin isn't also installed.

Reply:
We don't need a constant, just if class exists then.

Reply:
@snarfed I am not suggesting to remove the `MICROPUB_LOCAL_AUTH` constant literally, but to move the check to the micropub plugin instead of the indieauth plugin.

Reply:
I will amend the PR

Reply:
Another problem with the actual merge request: If `MICROPUB_LOCAL_AUTH` is defined with `0` it will add both headers.